spec/policies/pair_request_policy_spec.rb

```diff
@@ -0,0 +1,74 @@
+RSpec.describe PairRequestPolicy do
+  subject { described_class }
+
+  permissions :index?, :show? do
+    it 'denies access if the user is not the author, invitee, or an admin account' do
+      expect(subject).not_to permit(build(:user), build(:pair_request))
+    end
+
+    it 'grants access if the user is the author' do
+      authoring_user = build(:user)
+      expect(subject).to permit(authoring_user, build(:pair_request, author: authoring_user))
+    end
+
+    it 'grants access if the user is the invitee' do
+      invitee_user = build(:user)
+      expect(subject).to permit(invitee_user, build(:pair_request, invitee: invitee_user))
+    end
+
+    it 'grants access if the user is an admin' do
+      admin = build(:user, :admin)
+      expect(subject).to permit(admin, build(:pair_request))
+    end
+  end
+
+  permissions :new?, :create? do
+    it 'grants access to the user' do
+      expect(subject).to permit(build(:user), build(:pair_request))
+    end
+  end
+
+  permissions :accept? do
+    context 'when the user is not the invitee' do
+      it 'denies access' do
+        authoring_user = build(:user)
+        expect(subject).not_to permit(authoring_user, build(:pair_request, author: authoring_user))
```

Comment:
so for these types of tests generally what I do (and how I was taught) is you have a bunch of lets at the top (for like different types of users to use betweeen all the permissions). You can also include some `pair_requests` or scope those directly inside a permission. Typically in rspec you don't want to be building/creating objects but go with the `let` and `let!` approach (and sometimes `before do...`)

Let me know if that makes sense.

Reply:
It definitely makes sense, and I could write it that way.

This is one of those things that I regularly receive conflicting advice on lol. Some people openly dislike `let` variables in RSpec (including at least one of the seniors in the agency of learning) while others really like them.

But yeah, I figure the important thing is there's a consistent style in the tests, and if that's how you want it, I'll can definitely code it that way.